libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt

```diff
@@ -39,38 +36,204 @@ object MapboxDistanceUtil {
         context: Context,
         locale: Locale
     ): FormattedDistanceData {
-        return when (getFormattingRange(distanceInMeters, unitType)) {
-            FormattingRange.INVALID -> formatDistanceAndSuffixForSmallUnit(
+        val formattingData = getFormattingData(
+            distanceInMeters,
+            roundingIncrement,
+            unitType,
+            locale
+        )
+        val resources = context.applicationContext.resourcesWithLocale(locale)
+        val unitStringSuffix = getUnitString(resources, formattingData.turfDistanceUnit)
+        return FormattedDistanceData(
+            formattingData.distance,
+            formattingData.distanceAsString,
+            unitStringSuffix,
+            formattingData.unitType
+        )
+    }
+
+    private fun getFormattingData(
+        distanceInMeters: Double,
+        roundingIncrement: Int,
+        unitType: UnitType,
+        locale: Locale
+    ): FormattingData {
+        return when (unitType) {
+            UnitType.METRIC -> getMetricDistance(distanceInMeters, roundingIncrement, locale)
+            UnitType.IMPERIAL -> {
+                val distanceInMiles = TurfConversion.convertLength(
+                    distanceInMeters,
+                    TurfConstants.UNIT_METERS,
+                    TurfConstants.UNIT_MILES
+                )
+                if (locale.language == enLanguage && locale.country == "GB") {
+                    getUKDistance(distanceInMiles, roundingIncrement, locale)
+                } else {
+                    getUSDistance(distanceInMiles, roundingIncrement, locale)
```

Comment:
I'm not really an expert, but should we take into account other locations (like Australia and maybe some else)? Maybe set a more generic name for this? like `getYardDistance`. 
Also, do we really need to check the language for the GB locale?

Reply:
That's a good question. In the ticket only UK was being discussed.
@abhishek1508 @1ec5 are there any other countries that we should use yards in? As far as I understood, on iOS it's just GB, right?

Reply:
Checked that on iOS the logic is exactly the same: https://github.com/mapbox/mapbox-navigation-ios/blob/96d0fc0744d64d2a092f571e1583eeb86831a774/Sources/MapboxCoreNavigation/DistanceFormatter.swift#L140